dataland-frontend/src/components/general/ApiKeyCard.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <PrimeDialog
+    id="revokeModal"
+    :modal="true"
+    header="Header"
+    footer="Footer"
+    v-model:visible="viewDeleteConfirmation"
+    :breakpoints="{ '960px': '75vw', '640px': '100vw' }"
+    :style="{ width: '50vw' }"
+  >
+    <template #header>
+      <h2 class="m-0">Delete API Key</h2>
+    </template>
+    Are you sure you want to delete this API key?
+    <strong>If you confirm, your previous token will be invalidated and your applications will stop working.</strong>
+    <template #footer>
+      <PrimeButton label="CANCEL" @click="deleteConfirmToggle" class="p-button-outlined text-sm" />
+      <PrimeButton id="confirmRevokeButton" label="CONFIRM" @click="$emit('revokeKey')" class="text-sm" />
+    </template>
+  </PrimeDialog>
+
+  <div>
+    <div id="existingApiKeyCard" class="surface-card shadow-1 p-3 border-round-sm border-round">
+      <div class="flex justify-content-between mb-3">
+        <div>
+          <div class="text-900 font-medium text-xl text-left">API Key info</div>
+          <span :class="{ 'text-red-700': !isKeyExpired }" class="block text-600 mb-3 mt-6">
+            {{
+              isKeyExpired
+                ? `The API Key will expire on ${formatExpiryDate(expiryDateInDays)}`
+                : `The API Key expired ${formatExpiryDate(expiryDateInDays)}`
+            }}
+          </span>
+        </div>
+
+        <div>
+          <div class="text-left text-xs ml-1 text-600">Scope</div>
```

Comment:
The scopes should reflect the grants the user has - show READ only if user has role ROLE_USER, WRITE if he has ROLE_ADMIN

Reply:
done

Reply:
If a new API-Key is generated, the roles are not yet visible.
![image](https://user-images.githubusercontent.com/47553592/207320113-92e35516-f966-47c9-84cf-2f73ae895408.png)
![image](https://user-images.githubusercontent.com/47553592/207320181-1a303a28-6f15-48b6-9a0c-5d52c1c7de19.png)

Can you fix this? You should be able to get the roles from the keycloak-object.

Reply:
ok

Reply:
ok